it/content/src/main/content/jcr_root/content/forms/af/core-components-it/samples/numberinput/basic/.content.xml

```diff
@@ -1,53 +1,78 @@
-              id="tooltip_scenario_test"
-              description="This is long description"
-              tooltip="This is short description"
-              tooltipVisible="true"
-              sling:resourceType="forms-components-examples/components/form/numberinput"/>
+                    jcr:lastModified="{Date}2022-12-13T13:57:09.443+05:30"
+                    jcr:lastModifiedBy="admin"
+                    jcr:primaryType="nt:unstructured"
+                    jcr:title="Number Input 4"
+                    sling:resourceType="forms-components-examples/components/form/numberinput"
+                    description="This is long description"
+                    enabled="{Boolean}true"
+                    fieldType="number-input"
+                    id="tooltip_scenario_test"
+                    name="numberinput4"
+                    readOnly="{Boolean}false"
+                    textIsRich="[true,true]"
+                    tooltip="This is short description"
+                    tooltipVisible="true"
+                    type="number"
+                    validationExpression="$field == 64"
+                    visible="{Boolean}true">
+                <fd:rules
+                        fd:validate="[{&quot;nodeName&quot;:&quot;ROOT&quot;\,&quot;items&quot;:[{&quot;nodeName&quot;:&quot;STATEMENT&quot;\,&quot;choice&quot;:{&quot;nodeName&quot;:&quot;VALIDATE_EXPRESSION&quot;\,&quot;items&quot;:[{&quot;nodeName&quot;:&quot;AFCOMPONENT&quot;\,&quot;value&quot;:{&quot;id&quot;:&quot;$form.numberinput4&quot;\,&quot;type&quot;:&quot;AFCOMPONENT&quot;\,&quot;name&quot;:&quot;numberinput4&quot;}}\,{&quot;nodeName&quot;:&quot;Using&quot;\,&quot;value&quot;:null}\,{&quot;nodeName&quot;:&quot;Expression&quot;\,&quot;value&quot;:null}\,{&quot;nodeName&quot;:&quot;CONDITION&quot;\,&quot;choice&quot;:{&quot;nodeName&quot;:&quot;COMPARISON_EXPRESSION&quot;\,&quot;items&quot;:[{&quot;nodeName&quot;:&quot;EXPRESSION&quot;\,&quot;choice&quot;:{&quot;nodeName&quot;:&quot;COMPONENT&quot;\,&quot;value&quot;:{&quot;id&quot;:&quot;$form.numberinput4&quot;\,&quot;displayName&quot;:&quot;Number Input 4&quot;\,&quot;type&quot;:&quot;NUMBER&quot;\,&quot;displayPath&quot;:&quot;FORM/Number Input 4/&quot;\,&quot;name&quot;:&quot;numberinput4&quot;\,&quot;parent&quot;:&quot;$form&quot;}}}\,{&quot;nodeName&quot;:&quot;OPERATOR&quot;\,&quot;choice&quot;:{&quot;nodeName&quot;:&quot;EQUALS_TO&quot;\,&quot;value&quot;:null}}\,{&quot;nodeName&quot;:&quot;EXPRESSION&quot;\,&quot;choice&quot;:{&quot;nodeName&quot;:&quot;NUMERIC_LITERAL&quot;\,&quot;value&quot;:&quot;64&quot;}}]}\,&quot;nested&quot;:false}]}}]\,&quot;isValid&quot;:true\,&quot;enabled&quot;:true\,&quot;version&quot;:1\,&quot;script&quot;:&quot;$field == 64&quot;\,&quot;eventName&quot;:&quot;Validate&quot;\,&quot;ruleType&quot;:&quot;&quot;\,&quot;description&quot;:&quot;&quot;}]"
+                        jcr:primaryType="nt:unstructured"
+                        validationStatus="valid"/>
```

Comment:
jsonformula should also have been persisted.

Reply:
I just noticed that validationExpression is already saved here in json formula format. so we are good. (I forgot that we have separated validationExpression persistence out of fd:rules)